controllers/ccruntime_controller.go

```diff
@@ -235,6 +295,39 @@ func (r *CcRuntimeReconciler) processCcRuntimeInstallRequest() (ctrl.Result, err
 	return ctrl.Result{}, nil
 }
 
+func (r *CcRuntimeReconciler) handlePrePostDs(preInstallDs *appsv1.DaemonSet, doneLabel map[string]string) (
+	ctrl.Result, error,
+) {
+	foundPreinstallDs := &appsv1.DaemonSet{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: preInstallDs.Name, Namespace: preInstallDs.Namespace}, foundPreinstallDs)
+	if err != nil && errors.IsNotFound(err) {
+		err = r.Client.Create(context.TODO(), preInstallDs)
+		if err != nil {
+			r.Log.Info("failed to create preinstall/postuninstall DS")
+			return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 10}, err
+		}
+		return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 10}, nil
+	} else if err != nil {
+		r.Log.Info("couldn't GET preinstall/postuninstall DS")
+		return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 10}, err
+	}
+	// if ds exists, get all labels
+	err, nodes := r.getNodesWithLabels(doneLabel)
+	if err != nil {
+		r.Log.Info("couldn't GET labelled nodes")
+		return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 10}, err
+	}
+	if len(nodes.Items) < r.ccRuntime.Status.TotalNodesCount {
+		return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 10}, err
+	}
+	err = r.Client.Delete(context.TODO(), foundPreinstallDs)
```

Comment:
I'm wondering do we really need to delete the daemonsets. Any issues if we keep both the DS running to take care of node addition and removal to the cluster in a simpler way ?

Reply:
That's a good point, I'll include scaling up and down nodes into my tests. My guess is that it will be better to only delete as part of the CCRuntime finalizer.

Reply:
btw: I changed this now. The daemonsets stay running and are only deleted when the custom ressource is deleted altogether.